### DISCUSSION TITLE
diesel_cli generates CiText

### DISCUSSION THREAD

**Discussion by nicrosengren** [diesel-rs/diesel#3299] [Q&A] [answered]
Hi,

Before diesel 2.0.0 I used Citext by simply adding a simple type:

```
pub type Citext = diesel::sql_types::Text;
```

and then including this in my schema using diesel.toml's 

```
import_types = [ "crate::Citext", "diesel::sql_types::*" ]
```

But when upgrading to diesel 2.0.0 it seems diesel_cli generates it's own `Citext` in schema.rs. Is it possible to blacklist or in any other way disable this?

This unexpectedly became a major blocker. The diesel-citext crate is not an option.

**Comment by weiznich** [comment, marked as answer]
We do not provide an option to disable only specifc types. You can use a `diesel.toml` file (or the corresponding command line flag) to disable the generation of custom sql types at all. Checkout the [corresponding guide](http://diesel.rs/guides/configuring-diesel-cli.html) on our website for details.

Otherwise: PR's extending this functionality in a backward compatible way are welcome.

**Reply by nicrosengren** [reply]
I missed the `generate_missing_sql_type_definitions` option in diesel.toml. Thank you for pointing it out.

If you see `blacklisted_sql_types` array in `diesel.toml` or similar as a possible solution, I'd absolutely work on a PR for that.

Thanks for your reply!

**Reply by weiznich** [reply]
> I missed the generate_missing_sql_type_definitions option in diesel.toml. Thank you for pointing it out.

To be fair: I've just noticed today that this guide was not updated as part of the diesel 2.0 release and pushed an updated version of the web page.

> If you see blacklisted_sql_types array in diesel.toml or similar as a possible solution, I'd absolutely work on a PR for that.

That would be a fine solution as long as there are tests + documentation. PR's are always welcome.
